### PR TITLE
feat(filter): handle 'tag only' fields

### PR DESCRIPTION
**Pull request by rossigee**
Add ability to update fields that contain only an image tag, not a full image ref.

https://github.com/bluebrown/kobold/issues/19

### CODE REVIEW

**Comment by bluebrown** [comment]
Ah my man. This is exactly how I imagined it, a while back.

**Comment by rossigee** [comment]
Tested and working locally. Ready for wider review.

File: krm/filter.go [hunk 1]
```diff
@@ -40,6 +44,9 @@ func DefaultNodeHandler(_, currentRef, nextRef string, opts Options) (string, er
 		return currentRef, err
 	}
 
+	if opts.Part == "tag" {
```
Comment (bluebrown) [inline krm/filter.go:47]: I am not sure how to handle the tag in this situation. 

One way could be to have the digest as a seperate part to select from. Then we also need a way to combine parts into a single output field.

Or everything after the image is considered the tag or _identifier_ how the container package calls it.  

On top of that, the built in decoder actuall generate `<tag>@<digest>` identifiers. This is technically not correct. Registiries will ignore the tag but its nice for informational purposes.

**Comment by bluebrown** [comment]
I will merge this now. I think it can be as is, and I will add other parts called digest and identifier or something. Its actually already all there, I just need to switch on opts.part.

**Comment by bluebrown** [comment]
thanks for the work :)

**Pull request merged**